ti/instrumentation/csharp/nunit_console.go

```diff
@@ -0,0 +1,105 @@
+// Supports running tests via the nunit console test runner for C#
+//
+// Test filtering:
+// 		nunit3-console.exe <path-to-dll> --where "class =~ FirstTest || class =~ SecondTest"
+package csharp
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/harness/lite-engine/internal/filesystem"
+	"github.com/harness/lite-engine/ti"
+	"github.com/mholt/archiver/v3"
+
+	"github.com/sirupsen/logrus"
+)
+
+type nunitConsoleRunner struct {
+	fs  filesystem.FileSystem
+	log *logrus.Logger
+}
+
+func NewNunitConsoleRunner(log *logrus.Logger, fs filesystem.FileSystem) *nunitConsoleRunner { // nolint:revive
+	return &nunitConsoleRunner{
+		fs:  fs,
+		log: log,
+	}
+}
+
+func (b *nunitConsoleRunner) AutoDetectPackages(workspace string) ([]string, error) {
+	return []string{}, errors.New("not implemented")
+}
+
+func (b *nunitConsoleRunner) GetCmd(ctx context.Context, tests []ti.RunnableTest, userArgs, workspace, agentConfigPath, agentInstallDir string, ignoreInstr, runAll bool) (string, error) {
+	/*
+		i) Get the DLL list from the command (assume it runs at the root of the repository)
+		ii) Run the injector through all the DLLs
+		iii) Add test filtering
+
+		Working command:
+			. nunit3-console.exe <path-to-dll> --where "class =~ FirstTest || class =~ SecondTest"
+	*/
+	if ignoreInstr {
+		return userArgs, nil
+	}
+
+	var cmd string
+	pathToInjector := filepath.Join(agentInstallDir, "dotnet-agent", "dotnet-agent.injector.exe")
+
+	zip := archiver.Zip{
+		OverwriteExisting: true,
+	}
+	// Unzip everything at agentInstallDir/dotnet-agent.zip
+	err := zip.Unarchive(filepath.Join(agentInstallDir, "dotnet-agent.zip"), agentInstallDir)
+	if err != nil {
+		b.log.WithError(err).Println("could not unarchive the dotnet agent")
+		return "", err
+	}
+
+	// Run all the DLLs through the injector
+	args := strings.Split(userArgs, " ")
+	for _, s := range args {
+		if strings.HasSuffix(s, ".dll") {
+			absPath := filepath.Join(workspace, s)
```

Comment:
check if the path already begins with workspace in which case don't prepend workspace
This can happen when the user provides an absolute path